### DISCUSSION TITLE
Simple auth policy on update queries breaks GraphQL

### DISCUSSION THREAD

**Discussion by ian-s-mcb** [supabase/supabase#13618] [Questions] [answered]
I'm having some trouble restricting updates in my `profiles` table to only rows that belong to the current logged in user. My question is whether I made a mistake in my policy or whether I've misunderstood how GraphQL mutations are written.

With the following policy, updates to the `profiles` table have no effect, as in the GraphQL mutation results in an `affectedCount` of zero.
```sql
create policy "Users can update their own profile."
  on profiles for update
  using ( auth.uid() = id );
```

When replacing that policy with the following overly lenient one, my mutations succeed.

```sql
create policy "All users can update anybody's profie "
  on profiles for update
  using ( true );
```

```sql
-- My profiles table
create table profiles (
 id uuid not null references auth.users on delete cascade,
 display_name varchar(32),

 primary key (id)
);
alter table public.profiles enable row level security;
```

```graphql
# 1-Mutation query
mutation {
  updatedProfile: updateProfilesCollection(
    set: { displayName: "Junk" }
    filter: { id: { eq: "$ed5fc1e5-3be9-4cbc-bcfe-9ac2b746d32a" } }
  ) {
    affectedCount
    records {
      id
      displayName
    }
  }
}
# Results from #1
{
  "data": {
    "updatedProfile": {
      "records": [],
      "affectedCount": 0
    }
  }
}

# 2-Query to show original value
{
  profiles: profilesCollection(
    filter: {id: {eq: "ed5fc1e5-3be9-4cbc-bcfe-9ac2b746d32a"}}
  ) {
    edges {
      node {
        displayName
      }
    }
    }
  }
# Results from #2
{
  "data": {
    "profiles": {
      "edges": [
        {
          "node": {
            "displayName": "Abby's farm"
          }
        }
      ]
    }
  }
}
```

**Comment by ian-s-mcb** [comment, marked as answer]
This problem with updating records via GraphQL came down to insufficient headers in my GraphQL client.

I included my Supabase anon key in my headers but not the session access token. I'm not sure why my mutation didn't throw an error about row-level security policy. Either way, the Supabase docs for GraphQL need some expanding, especially for Supabase users migrating from the the official Client Libraries (like supabase-js).

It's great to see that Supabase Studio now has GraphiQL and I hope more improvements are on the way.

**Reply by ky1ejs** [reply]
@ian-s-mcb turns out that this is expected behavior of RLS for security reasons: https://github.com/orgs/supabase/discussions/15325#discussion-5331428
